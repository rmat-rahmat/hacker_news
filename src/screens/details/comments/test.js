/**
 * @format
 */

import React from 'react';
import 'react-native';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import Comment from '.';

describe('Test Comment screen', () => {


const mockComment =
  {
    "by": "tester2",
    "id": 20,
    "kids": [3],
    "parent": 1,
    "text": "this is comment 2",
    "time": 1314211127,
    "type": "comment"
  }

it('renders Comment correctly', () => {
  const tree = renderer.create(
    <Comment data={mockComment} />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
});