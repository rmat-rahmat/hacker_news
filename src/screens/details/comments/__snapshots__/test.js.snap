// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Comment screen renders Comment correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "flex": 0.9,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "grey",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      tester2
       || 
      10 years ago
    </Text>
    <View
      style={
        Array [
          Object {},
          undefined,
          undefined,
        ]
      }
      testID="html"
    >
      <View
        style={
          Array [
            Object {},
            undefined,
            undefined,
          ]
        }
        testID="body"
      >
        <Text
          allowFontScaling={true}
          selectable={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 15,
              },
              undefined,
              undefined,
            ]
          }
        >
          this is comment 2
        </Text>
      </View>
    </View>
    <View
      style={Object {}}
    />
  </View>
</View>
`;
